File: CodeChef/LazyJem.py
for _ in range(int(input())):
    a, b, c = list(map(int, input().split()))
    ans = 0
    while a > 0:
        if a % 2 == 1:
            solved = (a + 1) / 2
        else:
            solved = (a / 2)
        ans += solved * c
        a -= solved
        if a >= 0:
            ans += 1
        c *= 2
    print(int(ans))

File: CodeChef/LazyJem.cpp
#include <bits/stdc++.h>

typedef long long int lli;

using namespace std;

inline void solve() {
  lli a, b, c, ans = 0;
  cin >> a >> b >> c;
  while (a > 0) {
    lli solved;
    if (a % 2 == 0) solved = a / 2;
    else solved = (a + 1) / 2;
    ans += solved * c;
    a -= solved;
    if (a > 0)
      ans += b;
    c *= 2;
  }
  cout << ans << '\n';
}
int main() {
  int t;
  cin >> t;
  while (t--)
    solve();
}
